lib/segment/src/vector_storage/quantized/quantized_vectors.rs

```diff
@@ -1009,6 +1072,271 @@ impl QuantizedVectors {
+                        Self::load_mutable_binary_multi(binary, config, v.as_ref(), path)?,
+                    )),
+                    VectorStorageEnum::MultiDenseAppendableMemmapByte(v) => Ok(Some(
+                        Self::load_mutable_binary_multi(binary, config, v.as_ref(), path)?,
+                    )),
+                    VectorStorageEnum::MultiDenseAppendableMemmapHalf(v) => Ok(Some(
+                        Self::load_mutable_binary_multi(binary, config, v.as_ref(), path)?,
+                    )),
+                    VectorStorageEnum::MultiDenseAppendableInRam(v) => Ok(Some(
+                        Self::load_mutable_binary_multi(binary, config, v.as_ref(), path)?,
+                    )),
+                    VectorStorageEnum::MultiDenseAppendableInRamByte(v) => Ok(Some(
+                        Self::load_mutable_binary_multi(binary, config, v.as_ref(), path)?,
+                    )),
+                    VectorStorageEnum::MultiDenseAppendableInRamHalf(v) => Ok(Some(
+                        Self::load_mutable_binary_multi(binary, config, v.as_ref(), path)?,
+                    )),
+                }
+            }
+            _ => {
+                debug_assert!(
+                    false,
+                    "Only binary quantization is supported for mutable quantized vectors"
+                );
+                Ok(None)
+            }
+        }
+    }
+
+    fn load_mutable_binary_dense<
+        TElement: PrimitiveVectorElement,
+        TVectorStorage: DenseVectorStorage<TElement> + Send + Sync,
+    >(
+        binary_config: BinaryQuantizationConfig,
+        quantization_config: QuantizedVectorsConfig,
+        vector_storage: &TVectorStorage,
+        path: &Path,
+    ) -> OperationResult<Self> {
+        let data_path = Self::get_appendable_data_path(path);
+        let meta_path = Self::get_meta_path(path);
+
+        let on_disk_vector_storage = vector_storage.is_on_disk();
+        let in_ram = Self::is_ram(binary_config.always_ram, on_disk_vector_storage);
+        let advice = if in_ram {
+            AdviceSetting::from(Advice::Normal)
+        } else {
+            AdviceSetting::Global
+        };
+
+        let encoding = Self::convert_binary_encoding(binary_config.encoding);
+        let query_encoding = Self::convert_binary_query_encoding(binary_config.query_encoding);
+        let quantized_vector_size =
+            EncodedVectorsBin::<u128, QuantizedMmapStorage>::get_quantized_vector_size_from_params(
+                quantization_config.vector_parameters.dim,
+                encoding,
+            );
+        let quantization_storage = ChunkedMmapVectors::<u8>::open(
+            data_path.as_path(),
+            quantized_vector_size,
+            Some(false), // mlock
+            advice,
+            Some(in_ram), // populate
+        )?;
+
+        // If quantization is not present, create a new one
+        let storage_impl = if meta_path.exists() {
```

Comment:
Not sure if such a check is correct. Instead, I can add `create: bool` argument to the function, on a higher level, I know already, do I perform loading or creation